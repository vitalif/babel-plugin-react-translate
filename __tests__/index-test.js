const fs = require('fs');
const babel = require('@babel/core');
const jsx = require('@babel/plugin-transform-react-jsx');
const plugin = require('../');

var example = '\
var name = "John";\
var foo = "Ё-Mobile";\
foo = "Меня зовут "+name+"!  ";\
foo = `  Меня зовут ${name}!`;\
foo = <span>Привет!</span>;\
foo = L("У меня много {1}: ", L("братьев"));\
var age = { "Братья": { "Иван": "Горький", "Mikhail": "Золотухин" }, ["Сёстры"+getnum()]: {}, Родители };\
';

it('works', () =>
{
    const { code } = babel.transform(example, { filename: 'test', plugins: [ jsx, plugin ] });
    expect(code).toMatchSnapshot();
    expect(JSON.parse(fs.readFileSync('react-translate-output.json'))).toEqual({
        test: [
            "Ё-Mobile",
            "Меня зовут {1}!",
            "Привет!",
            "У меня много {1}: ",
            "братьев",
            "Братья",
            "Иван",
            "Горький",
            "Золотухин",
            "Сёстры{1}"
        ]
    });
});
