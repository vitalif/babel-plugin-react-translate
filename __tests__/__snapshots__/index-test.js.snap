// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"import { L } from \\"babel-plugin-react-translate/runtime\\";
var name = \\"John\\";
var foo = L(\\"\\\\u0401-Mobile\\");
foo = L(\\"\\\\u041C\\\\u0435\\\\u043D\\\\u044F \\\\u0437\\\\u043E\\\\u0432\\\\u0443\\\\u0442 {1}!\\", name) + \\"  \\";
foo = \\"  \\" + L(\\"\\\\u041C\\\\u0435\\\\u043D\\\\u044F \\\\u0437\\\\u043E\\\\u0432\\\\u0443\\\\u0442 {1}!\\", name);
foo = /*#__PURE__*/React.createElement(\\"span\\", null, L(\\"\\\\u041F\\\\u0440\\\\u0438\\\\u0432\\\\u0435\\\\u0442!\\"));
foo = L(\\"У меня много {1}: \\", L(\\"братьев\\"));"
`;
